vmdb/app/controllers/miq_request_controller.rb

```diff
@@ -426,112 +426,47 @@ def get_request_tab_type
-        end
-      end
-      cond.push(@or_hash)
+    if (a_s = opts[:applied_states].presence)
+      cond.push("or" => a_s.collect { |s| {"=" => {"value" => s, "field" => "MiqRequest-approval_state"}} })
     end
 
-    # Add time condition
-    cond_hash = Hash.new
-    cond_hash["AFTER"] = {"value"=>"#{opts[:time_period].to_i} Days Ago","field"=>"MiqRequest-created_on"}
-    cond.push(cond_hash)
-
-    case @layout
-    when "miq_request_ae"
-      req_typ = :AutomationRequest
-    when "miq_request_host"
-      req_typ = :Host
-    else
-      req_typ = :Vm
-    end
-    request_types = MiqRequest::MODEL_REQUEST_TYPES[req_typ]
-    request_types.each_with_index do |typ,i|
-      typ.each do |k|
-        if k.class == Symbol
-          if i == 0
-            @or_hash = Hash.new   # need this  incase there are more than one type
-            @or_hash["or"] = Array.new
-            cond_hash = Hash.new
-            cond_hash["="] = {"value"=>k.to_s,"field"=>"MiqRequest-resource_type"}
-            @or_hash["or"].push(cond_hash)
-          elsif i == request_types.length-1
-            cond_hash = Hash.new
-            cond_hash["="] = {"value"=>k.to_s,"field"=>"MiqRequest-resource_type"}
-            @or_hash["or"].push(cond_hash)
-          else
-            cond_hash = Hash.new
-            cond_hash["="] = {"value"=>k.to_s,"field"=>"MiqRequest-resource_type"}
-            @or_hash["or"].push(cond_hash)
-          end
-        end
+    req_typ =
+      case @layout
+      when "miq_request_ae"   then :AutomationRequest
+      when "miq_request_host" then :Host
+      else                         :Vm
       end
-    end
-    cond.push(@or_hash)
+    types = MiqRequest::MODEL_REQUEST_TYPES[req_typ]
+    cond.push("or" => types.keys.collect { |k| {"=" => {"value" => k.to_s, "field" => "MiqRequest-resource_type"}} })
 
     if opts[:type_choice] && opts[:type_choice] != "all"  # Add request_type filter, if selected
-      cond_hash = Hash.new
-      cond_hash["="] = {"value"=>opts[:type_choice],"field"=>"MiqRequest-request_type"}
-      cond.push(cond_hash)
+      cond.push("=" => {"value" => opts[:type_choice], "field" => "MiqRequest-request_type"})
     end
 
-    if opts[:reason_text]  && opts[:reason_text] != ""
-      cond_hash = Hash.new
-      if opts[:reason_text].starts_with?("*") && opts[:reason_text].ends_with?("*")   # Replace beginning/ending * chars with % for SQL
-        hash_key = "INCLUDES"
-        reason_text = opts[:reason_text][1..-2]
-      elsif opts[:reason_text].starts_with?("*")
-        hash_key = "ENDS WITH"
-        reason_text = opts[:reason_text][1..-1]
-      elsif opts[:reason_text].ends_with?("*")
-        hash_key = "STARTS WITH"
-        reason_text = opts[:reason_text][0..-2]
-      else
-        hash_key = "INCLUDES"
-        reason_text = opts[:reason_text]
-      end
+    if (text = opts[:reason_text].presence)
+      new_text = text.gsub(/\A\*/, "").chomp("*")  # Remove leading and/or trailing "*"
```

Comment:
1. Can you pull this out to another method?
2. Feels like it is doing too many regexs.

I have an alternative, though I don't necessarily like it more than what you have.
Not sure if this new code would work with multiple stars either.
Does this give you any ideas for improving the code you wrote?

```ruby
text =~ /(\*?)([^*]+)(\*?)/
new_text = $2

hash_key =
  case $1 + $3
  when "* " then "STARTS WITH"
  when " *" then "ENDS WITH"
  else            "INCLUDES"
end
```